message_tagging_service/tagging_service.py

```diff
@@ -311,15 +327,34 @@ def handle(rule_defs, event_msg):
         for rule_def in rule_defs
     )))
 
-    if rule_matches:
-        stream = this_stream.replace('-', '_')
-        nvr = f'{this_name}-{stream}-{this_version}.{this_context}'
-        dest_tags = [item.dest_tag for item in rule_matches]
-        logger.debug('Tag build %s with tag(s) %s', nvr, ', '.join(dest_tags))
-        if mts_conf.dry_run:
-            logger.info('DRY-RUN: tag build nvr: %s, destination tags: %r',
-                        nvr, dest_tags)
-        else:
-            tag_build(nvr, dest_tags)
-    else:
+    if not rule_matches:
         logger.info('Module build %s does not match any rule.', nsvc)
+        return
+
+    stream = this_stream.replace('-', '_')
+    nvr = f'{this_name}-{stream}-{this_version}.{this_context}'
+    dest_tags = [item.dest_tag for item in rule_matches]
+    logger.debug('Tag build %s with tag(s) %s', nvr, ', '.join(dest_tags))
+    if mts_conf.dry_run:
+        logger.info('DRY-RUN: tag build nvr: %s, destination tags: %r',
+                    nvr, dest_tags)
+    else:
+        tagged_tags = tag_build(nvr, dest_tags)
+
+        if not tagged_tags:
+            logger.warning(
+                'None of tag(s) %r is tagged to build %s. Skip to send message.',
```

Comment:
I wonder if we should still send message in this case. Otherwise, consumer of this message may not know whether tags weren't applied, or MTS hasn't had a chance to process the build yet.

Reply:
This should be a temporary solution at this moment. `tagBuild` just creates a task, and there is an issue for sending message only when task is done. I think we can consider this in a new consumer of particular event message for that issue.